### DISCUSSION TITLE
View Code Coverage of Current Code Changes

### DISCUSSION THREAD

**Discussion by csteague7** [FortuneN/FineCodeCoverage#303] [Ideas]
Sonar Qube integrated with code coverage allows us to see the line coverage of pull request changes.  It shows an overall percentage, and provides the ability to click a link and view the list of files with uncovered lines in the pr.

It would be great if Fine Code Coverage provided the same feature for Current Code Changes.  This will allow the developer to meet our required code coverage matrix without first submitting the code as a pr.  This is important because the c# solution that we would use this on first is very large (500K +), with a large number of projects.  So, it takes a long time to build.

@tonyhallett I see that you answered many posts well. What do you think of the above?

**Comment by csteague7** [comment]
I am asking Microsoft for help with this.
https://developercommunity.visualstudio.com/t/Resetting-Severity-of-Compiler-error-Fai/10190813?

**Comment by tonyhallett** [comment]
> What do you think of the above?

Sure it can be incorporated at some point.
I assume that FCC will run the appropriate git command to get the lines changed and to then filter the cobertura results to just these lines.  This information could then be displayed in the Summary tab as additional PR Covered Lines / PR Uncovered Lines / PR Line Coverage.  We could also take a coverage threshold option so as to colour red when insufficient lines covered.

Open to a pull request on this as I do not know when I can commit to this feature.

**Reply by csteague7** [reply]
Thank you, @tonyhallett .  I plan to work on the PR beginning in about month.  I will pair program with one of my colleagues.  Our intention is to finish during or before January.
